federation/blocksigning.py

```
@@ -7,22 +7,36 @@
 from .connectivity import getoceand
 
 class BlockSigning(DaemonThread):
-    def __init__(self, ocean_conf, messenger_type, nodes, my_id, block_time, signer=None):
+    def __init__(self, ocean_conf, messenger_type, nodes, my_id, block_time, in_rate, in_period, in_address, script, signer=None):
         super().__init__()
         self.ocean_conf = ocean_conf
         self.ocean = getoceand(self.ocean_conf)
         self.interval = block_time
         self.total = len(nodes)
         self.my_id = my_id % self.total
         self.messenger = MessengerFactory.get_messenger(messenger_type, nodes, self.my_id)
+        self.rate = in_rate
+        self.period = in_period
+        self.address = in_address
+        self.script = script
         self.signer = signer
+        if in_rate > 0:
+            try:
+                self.ocean.importprivkey(ocean_conf["reissuanceprivkey"])
+            except:
```

I think we should throw an exception here so that if there is an issue with the key it doesn't go further.